### PR TITLE
Schema Evolution for Metadata Dataset and clean up interfaces

### PR DESCRIPTION
**Pull request by prasar-ashutosh**
#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java [hunk 1]
```diff
@@ -324,14 +342,67 @@ public List<DatasetFilter> getLatestStagingFilters(RelationalConnection connecti
...
+        if (this.executor == null)
+        {
+            throw new IllegalStateException("Executor not initialized, call init(Connection) before invoking this method!");
+        }
+        // Validation: init(Datasets) must have been invoked
+        if (!this.datasetsInitialized)
+        {
+            throw new IllegalStateException("Datasets not initialized, call init(Datasets) before invoking this method!");
+        }
+    }
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java]: Can change the method names in the comments and the error messages to be more accurate: initExecutor and initDatasets respectively

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java [hunk 2]
```diff
@@ -435,7 +506,7 @@ private List<IngestorResult> performFullIngestion(RelationalConnection connectio
         }
 
         // Evolve Schema
-        evolveSchema();
+        evolveMainDatasetSchema();
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java]: we should call evolve() here to trigger both

Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java]: One problem here tho - if someone calls performFullIngestion, it seems like we will not record down the schema evol SQLs for metadata table in the IngestorResult

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java (hunk 1 above)
Comment (prasar-ashutosh) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java]: Sure, will change

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java (hunk 2 above)
Comment (prasar-ashutosh) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java]: Yes, will invoke evolve and return the results in the IngestorResult

**Pull request merged**